superset/db_engine_specs/postgres.py

```diff
@@ -78,8 +78,13 @@ def get_allow_cost_estimate(cls, extra: Dict[str, Any]) -> bool:
 
     @classmethod
     def estimate_statement_cost(cls, statement: str, cursor: Any) -> Dict[str, Any]:
+        from psycopg2 import errors
+
         sql = f"EXPLAIN {statement}"
-        cursor.execute(sql)
+        try:
+            cursor.execute(sql)
+        except errors.SyntaxError as err:
```

Comment:
I added handling exception but now I'm wondering if leaving it to be caught by errorhandler would be a better option.

Reply:
I think so too

Reply:
done